subworkflows/local/bam_qc/main.nf

```diff
@@ -9,58 +9,59 @@ include { PICARD_COLLECTWGSMETRICS      } from '../../../modules/nf-core/picard/
 
 workflow BAM_QC {
     take:
-        ch_bam_bai_target   // channel: [mandatory] [meta, bam, bai, target]
-        ch_fasta_fai        // channel: [mandatory] [meta2, fasta, fai]
-        ch_fasta_dict       // channel: [mandatory] [meta2, dict]
-        disable_picard      // boolean: [optional]  [true]
+        ch_meta_reads_index   // channel: [mandatory] [meta, bam, bai]
+        disable_picard        // boolean: [optional]  [true]
 
     main:
         ch_versions = Channel.empty()
         ch_metrics  = Channel.empty()
 
-        ch_bait_interval_list = []
-        ch_target_interval_list = []
-
-        ch_meta_fai   = ch_fasta_fai.map {meta, fasta, fai -> [meta, fai]  }.collect()
-        ch_meta_fasta = ch_fasta_fai.map {meta, fasta, fai -> [meta, fasta]}.collect()
-
-        ch_bam_bai = ch_bam_bai_target.map{ meta, bam, bai, bed -> [meta, bam, bai]}
+        ch_meta_reads_index
+        | multiMap { meta, reads, index ->
+            meta_reads_index:   [meta, reads, index]
+            meta_fasta:         [meta, WorkflowMain.getGenomeAttribute(meta.genome, 'fasta')]
+            meta_fai:           [meta, WorkflowMain.getGenomeAttribute(meta.genome, 'fai')]
+            meta_dict:          [meta, WorkflowMain.getGenomeAttribute(meta.genome, 'dict')]
```

Comment:
Are these actually files or strings of the filenames? This could maybe cause some staging problems down the line?

Reply:
These are string values that are converted to `file` objects in the `getGenomeAttribute` function.
I'm not sure, but I haven't noticed any staging issues as of now

Reply:
Allright, I just wasn't sure :)